src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/ElfObjectWriter.cs

```diff
@@ -567,6 +574,9 @@ private void EmitRelocationsRiscV64(int sectionIndex, List<SymbolicRelocation> r
 
         private protected override void EmitSectionsAndLayout()
         {
+            SectionWriter commentSectionWriter = GetOrCreateSection(CommentSection);
+            commentSectionWriter.WriteUtf8String($".NET AOT {Assembly.GetExecutingAssembly().GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion}");
```

Comment:
I assume the informational version is what we want here. I don’t know what else we would pick. It might be nice to include the git sha, but I’m not sure if we put that anywhere but the PDB. Jeremy do you know offhand?

Reply:
InformationVersion is a good choice. It returns semver with semver 'metadata', like  e.g. `10.0.0-preview.3.25152.4+088d199063dd1bf7fa00a445d23f93f915f84b31`. That's what differenciates it from `Environment.Version`. https://github.com/dotnet/runtime/blob/f6948fca8fb88491f1c938169140736fc85c4d11/src/libraries/System.Private.CoreLib/gen/ProductVersionInfoGenerator.cs#L11

cc @akoeplinger who was worked on similar topic recently (related to servicing versions)

Reply:
InformationVersion is a good choice. It returns semver with semver 'metadata', like  e.g. `10.0.0-preview.3.25152.4+088d199063dd1bf7fa00a445d23f93f915f84b31`. That's what differenciates it from `Environment.Version` (which strips away the metadata). https://github.com/dotnet/runtime/blob/f6948fca8fb88491f1c938169140736fc85c4d11/src/libraries/System.Private.CoreLib/gen/ProductVersionInfoGenerator.cs#L11

cc @akoeplinger who worked on similar topic recently (related to servicing versions)

Reply:
The informational version has the commit sha (as @am11 points out), so I think it's the right choice for the scenario.